### ISSUE TITLE
Publishing the site to asciidoc-wg.eclipse.org

### ISSUE THREAD

**Issue by ahus1** [EclipseFdn/asciidoc-wg.eclipse.org#16]
This will be done once all content is ready for the first publishing of this site.

All content that should be part of this is marked as Milestone "M1"

Pre-requisite: #3 (rename from asciidoc to asciidoc-wg)

**Comment by ahus1** [comment]
Hi @chrisguindon - we plan to have the initial content ready early next week. 

What would we need to do once the content is there to get it published at asciidoc-wg.eclipse.org? (please not, the name has changed, see #3 for details)

**Comment by chrisguindon** [comment]
I've made the repo name change now.

We will want to host this website on Eclipse Infra since it's a subdomain of eclipse.org. To do so, we will need to commit our k8s configs similar to (https://github.com/EclipseFdn/iot.eclipse.org/tree/master/k8s), a Dockerfile (https://github.com/EclipseFdn/iot.eclipse.org/blob/master/Dockerfile), and a Jenkinsfile https://github.com/EclipseFdn/iot.eclipse.org/blob/master/Jenkinsfile.

This is usually straightforward, however, I believe we plan on using AsciiDoc format with Hugo: 
https://gohugo.io/content-management/formats/

Is that still true?

**Comment by ahus1** [comment]
Hi @chrisguindon - thanks for the name change. I merged the PR and it works great. 

Yes, the plan is to use AsciiDoc for the content. Hugo is already set up for this. From what I read, Hugo uses the Asciidoctor CLI during conversion. 

For Netlify I 
* added a [`Gemfile`](https://github.com/EclipseFdn/asciidoc-wg.eclipse.org/blob/main/Gemfile) to have Asciidoctor installed. 
* called `bundle` before calling npm and Hugo - see `netlify.toml`
* added `--destination public` to the Hugo command - see `netlify.toml` again

I'll give it a go and prepare some file. At least the Dockerfile I should be able to test locally. For all other files I'll probably need some help.

**Comment by chrisguindon** [comment]
I will need to create a project on our Jenkins instance once the files are in the repo.

**Comment by ahus1** [comment]
@chrisguindon - I've added the files you asked for (Dockerfile, Jenkinsfile, k8s) taking the IOT files as a template. 

I tested the Dockerfile locally and it will build a static site with Hugo and Asciidoctor. 

Could you please take it from here with the Jenkins job and the Kubernetes resources? I assume it is ready for staging. 

To be ready for production we'll need to complete the content first.

**Comment by chrisguindon** [comment]
I created a project in our Jenkins instance. I also have some patches ready for review for nginx.

@mbarbero can you let us know if we are missing anything before we can deploy the site?

@ahus1 I expect both staging and production to be ready at the same time.

**Comment by ahus1** [comment]
@chrisguindon - a short update: All initial content is now in place. We're ready to publish to asciidoc-wg.eclipse.org (both production and staging).

It would be great if you and @mbarbero could have a look after the Easter holidays to take the next steps. 

Thanks!

**Comment by Mogztter** [comment]
The build on Jenkins is currently failing (might be "normal"):

![image](https://user-images.githubusercontent.com/333276/113489470-c5533e80-94c4-11eb-93fe-f49ca9cc699a.png)

Also, I get an "Access denied" on Jenkins (I'm using my Eclipse account/email) so I cannot see the logs:

![image](https://user-images.githubusercontent.com/333276/113489496-e0be4980-94c4-11eb-8887-9f19fb57a66b.png)

Who should I ask to get access?
Thanks!

**Comment by ahus1** [comment]
@Mogztter IMHO Jenkins is about publishing the site to eclipse.org and it is work in progress. As long as Netlify is fine you are good to go.

**Comment by chrisguindon** [comment]
@ahus1 The Jenkins instance is only available to foundation staff. 

@mbarbero is there a way to make the logs public?

**Comment by chrisguindon** [comment]
I've also ping @mbarbero to see if he needs me to do anything else before he can make this live.

**Comment by mbarbero** [comment]
Everything should be in place right now. I've created the dns entry for https://asciidoc-wg.eclipse.org so it should start becoming available in about 15min (and can take up to much longer depending on DNS replication timing).

@chrisguindon, regarding permission to Jenkins, we can grant access to individual, but it's a maintenance burden (you need to keep track manually of who get access to what in multiple locations, and revoking access is a pain...). What about creating groups in LDAP? This way you could also handle teams at github automatically.

**Comment by ahus1** [comment]
Hi @chrisguindon, in #40 you commented that it "should work now". 

I tried this without success: 

```
curl https://asciidoc-wg.eclipse.org/ -v
...
HTTP/1.1 301 Moved Permanently
Location: https://accounts.eclipse.org/
```

I tried calling the new URL, but it redirects to accounts. Could you please re-check? I also tried to push on change to main, and that failed somewhere in Jenkins. 

Thanks!

**Comment by mbarbero** [comment]
I've fixed https://asciidoc-wg.eclipse.org. Please try again.

**Comment by ahus1** [comment]
After I persuaded my browser that was persistently caching the redirect (probably due to the 301): 

It works, whoot! 🎉 

Regarding access to the Jenkins logs: as long as they are green and we don't change/fix the Jenkinsfile I think we're fine, so for me it would be nice-to-have.
